lib/mix/tasks/eunit.ex

```diff
@@ -76,15 +89,39 @@ defmodule Mix.Tasks.Eunit do
                    _ -> []
                  end
 
-    %{eunit_opts: eunit_opts, patterns: patterns, cover: switches[:cover]}
+    %{eunit_opts: eunit_opts,
+      patterns: patterns,
+      profile: switches[:profile],
+      nocolor: switches[:no_color],
+      cover: switches[:cover]}
   end
 
   defp eunit_post_config(existing_config) do
     [erlc_paths: existing_config[:erlc_paths] ++ ["test"],
      erlc_options: existing_config[:erlc_options] ++ [{:d, :TEST}],
-     eunit_opts: existing_config[:eunit_opts]]
+     eunit_opts: existing_config[:eunit_opts] || []]
   end
 
+  defp get_eunit_opts(options, post_config) do
+    eunit_opts = options[:eunit_opts] ++ post_config[:eunit_opts]
+    maybe_add_formatter(eunit_opts, options[:profile], options[:nocolor])
+  end
+
+  defp maybe_add_formatter(opts, profile, nocolor) do
+    if List.keymember?(opts, :report, 0) do
```

Comment:
No real reason. I modeled this after similar code in rebar3, and it didn't occur to me to use `Keyword`. I will change it, `Keyword` is clearer.